lib/modem_info/modem_info.c

```diff
@@ -447,18 +456,37 @@ int modem_info_string_get(enum modem_info info, char *buf)
 			  CONFIG_MODEM_INFO_BUFFER_SIZE,
 			  NULL);
 
-	/* modem_info does not yet support array objects, so here we handle
-	 * the supported bands independently as a string
-	 */
+	/* modem_info does not yet support array objects, so here we handle */
+	/* the supported bands independently as a string */
 	if (info == MODEM_INFO_SUP_BAND) {
 		strcpy(buf, recv_buf + sizeof("%XCBAND: ") - 1);
 		return strlen(buf);
 	}
+	if (info == MODEM_INFO_IP_ADDRESS) {
+		/* check for multiple IP addresses */
+		while ((ip_str_end = strstr(ip_str_end, AT_CMD_RSP_DELIM))
+			   != NULL) {
+			++ip_str_end;
+			++ip_cnt;
+		}
+		LOG_DBG("Device contains %d IP addresses", ip_cnt);
+	}
 
 	if (err != 0) {
 		return -EIO;
 	}
 
+parse:
+	if (info == MODEM_INFO_IP_ADDRESS) {
+		/* parse each IP address line separately */
+		ip_str_end = strstr(&recv_buf[cmd_length], AT_CMD_RSP_DELIM);
+		if (ip_str_end == NULL) {
+			return -EFAULT;
+		}
+		/* get the size and then null-terminate the line */
+		ip_str_len = ip_str_end - &recv_buf[cmd_length];
+		recv_buf[++ip_str_len] = 0;
```

Comment:
How concerned should we be here that bad input data could result in a buffer overrun?  If ip_str_len becomes larger than the size of recv_buf[], bad things happen.